t/mojo/template.t

```diff
@@ -20,170 +20,202 @@ sub exception { die 'ohoh' }
+  my $output = $mt->render('    <%= "one" =%><%= "two" %>  three');
+  is $output, "onetwo  three\n", 'expression tags trimmed';
+};
+
+subtest 'Nothing to trim' => sub {
+  my $mt     = Mojo::Template->new;
+  my $output = $mt->render('<% =%>');
+  is $output, '', 'nothing trimmed';
+};
+
+subtest 'Replace tag' => sub {
+  my $mt     = Mojo::Template->new;
+  my $output = $mt->render('<%% 1 + 1 %>');
+  is $output, "<% 1 + 1 %>\n", 'tag has been replaced';
+};
+
+subtest 'Replace expression tag' => sub {
+  my $mt     = Mojo::Template->new;
+  my $output = $mt->render('<%%= 1 + 1 %>');
+  is $output, "<%= 1 + 1 %>\n", 'expression tag has been replaced';
+};
+
+subtest 'Replace expression tag (alternative)' => sub {
+  my $mt     = Mojo::Template->new;
+  my $output = $mt->render(' lalala <%%= 1 + 1 %> 1234 ');
+  is $output, " lalala <%= 1 + 1 %> 1234 \n", 'expression tag has been replaced';
+};
+
+subtest 'Replace expression tag (another alternative)' => sub {
+  my $mt     = Mojo::Template->new;
+  my $output = $mt->render(<<EOF);
 lalala <%%= 1 +
  1 %> 12
 34
 EOF
-is $output, "lalala <%= 1 +\n 1 %> 12\n34\n", 'expression tag has been replaced';
+  is $output, "lalala <%= 1 +\n 1 %> 12\n34\n", 'expression tag has been replaced';
+};
 
-# Replace comment tag
-$mt     = Mojo::Template->new;
-$output = $mt->render('<%%# 1 + 1 %>');
-is $output, "<%# 1 + 1 %>\n", 'comment tag has been replaced';
+subtest 'Replace comment tag' => sub {
+  my $mt     = Mojo::Template->new;
+  my $output = $mt->render('<%%# \'1 + 1 %>');
+  is $output, "<%# '1 + 1 %>\n", 'comment tag has been replaced';
+};
 
-# Replace line
-$mt     = Mojo::Template->new;
-$output = $mt->render('%% my $foo = 23;');
-is $output, "% my \$foo = 23;\n", 'line has been replaced';
+subtest 'Replace line' => sub {
+  my $mt     = Mojo::Template->new;
+  my $output = $mt->render('%% my $foo = 23;');
+  is $output, "% my \$foo = 23;\n", 'line has been replaced';
+};
 
-# Replace expression line
-$mt     = Mojo::Template->new;
-$output = $mt->render('  %%= 1 + 1');
-is $output, "  %= 1 + 1\n", 'expression line has been replaced';
+subtest 'Replace expression line' => sub {
+  my $mt     = Mojo::Template->new;
+  my $output = $mt->render('  %%= 1 + 1');
+  is $output, "  %= 1 + 1\n", 'expression line has been replaced';
+};
 
-# Replace expression line (alternative)
-$mt     = Mojo::Template->new;
-$output = $mt->render('%%= 1 + 1');
-is $output, "%= 1 + 1\n", 'expression line has been replaced';
+subtest 'Replace expression line (alternative)' => sub {
+  my $mt     = Mojo::Template->new;
+  my $output = $mt->render('%%= 1 + 1');
+  is $output, "%= 1 + 1\n", 'expression line has been replaced';
+};
 
-# Replace comment line
-$mt     = Mojo::Template->new;
-$output = $mt->render('  %%# 1 + 1');
-is $output, "  %# 1 + 1\n", 'comment line has been replaced';
+subtest 'Replace comment line' => sub {
+  my $mt     = Mojo::Template->new;
+  my $output = $mt->render('  %%# \'1 + 1');
```

Comment:
Addition of `'` to a test input and expected response. Test still passes, but unsure of the reason.